### PR TITLE
feat: truncation with custom ellipsis

### PR DESCRIPTION
**Pull request by sdaqo**
This addresses #131 by truncating the text and adding the ability to set an custom ellipsis.

### CODE REVIEW

**Comment by sdaqo** [comment]
@pavdmyt Sorry for the bump, would like to know if there is still anything to do here.

File: yaspin/core.py [hunk 1]
```diff
@@ -436,6 +447,16 @@ def _compose_color_func(self) -> Optional[Callable[..., str]]:
     def _compose_out(self, frame: str, mode: Optional[str] = None) -> str:
         text = str(self._text)
 
+        term_width = shutil.get_terminal_size()[0]
```
Comment (pavdmyt) [inline yaspin/core.py:450]: nit: let's use `shutil.get_terminal_size().columns` for clarity

File: yaspin/core.py [hunk 2]
```diff
@@ -436,6 +447,16 @@ def _compose_color_func(self) -> Optional[Callable[..., str]]:
...
+
+        # Width - (Frame + Space) - (Timer + Space) - Ellipsis
+        max_text_len = term_width - (len(frame) + 1)
+        max_text_len = max_text_len - (0 if not self._timer else 12 + 1)
```
Comment (pavdmyt) [inline yaspin/core.py:454]: Where is this `12 + 1` comes from? If this is a timer formatter from:
```
            text += " ({}.{:02.0f})".format(  # pylint: disable=consider-using-f-string
                timedelta(seconds=sec), fsec
            )
```
let's abstract it away into a dedicated variable and apply `len()` on it here.

**Review by pavdmyt — COMMENTED** [review]
@sdaqo Thanks for your contribution! Tried running your example, looks good 👍🏼 

Left several clarification comments. Please also add some unittests. The task can be simplified if you abstract away the logic for calculating `max_text_len` into a separate function that returns int value and writing some tests expecting various `frame` length, `timer` on/off, etc.

And apologies for the late response, I've been on a long vacation.

**Comment by pavdmyt** [comment]
Also, there should be an entry about `ellipsis` argument in `yaspin/api.py` (in the docstring for the `yaspin` function).

**Comment by sdaqo** [comment]
Alright, this should address everything, let me know if I left something out!

**Comment by pavdmyt** [comment]
CI `lint` step fails with:
```
ERROR: /home/runner/work/yaspin/yaspin/tests/test_ellipsis.py Imports are incorrectly sorted and/or formatted.
make: *** [Makefile:31: check-fmt] Error 1
Error: Process completed with exit code 2.
```

You can run `make check-fmt` to check formatting for your changes and `make isort`, `make fmt` to properly format your changes.

**Comment by sdaqo** [comment]
Wait I will clean the lint errors...

**Comment by sdaqo** [comment]
ok, fixed and squashed

**Review by pavdmyt — APPROVED** [review]

**Comment by pavdmyt** [comment]
@sdaqo Thanks for your contribution 👍🏼  This PR will be included in the next yaspin release.

**Pull request merged**